x-pack/plugins/endpoint/public/applications/endpoint/view/alerts/details/overview/take_action_dropdown.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useMemo, useState } from 'react';
+import { EuiPopover, EuiFormRow, EuiButton, EuiButtonEmpty } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+export const TakeActionDropdown = memo(() => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const TakeActionButton = useMemo(() => {
+    return (
+      <EuiButton
+        iconType="arrowDown"
+        iconSide="right"
+        data-test-subj="alertDetailTakeActionDropdownButton"
+        onClick={() => setIsDropdownOpen(!isDropdownOpen)}
+      >
+        <FormattedMessage
+          id="xpack.endpoint.application.endpoint.alertDetails.takeAction.title"
+          defaultMessage="Take Action"
+        />
+      </EuiButton>
+    );
+  }, [isDropdownOpen]);
+
+  return (
+    <EuiPopover
+      button={TakeActionButton}
+      isOpen={isDropdownOpen}
+      anchorPosition="downRight"
+      closePopover={() => setIsDropdownOpen(false)}
+      data-test-subj="alertListTakeActionDropdownContent"
+    >
+      <EuiFormRow>
+        <EuiButtonEmpty
+          data-test-subj="alertDetailTakeActionCloseAlertButton"
+          color="text"
+          onClick={() => {}}
```

Comment:
Can you remove this function? It shouldn't do anything, and having a new function reference on each render will force `EuiButtonEmpty` to rerender each time